File: modulo1/src/ros2_ws/src/fulgor_py_examples/fulgor_py_examples/subscriber.py
#Descripción: Nodo suscriptor del tópico /información que recibe mensajes del tipo String.

#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

#definición del nodo
class Subscriber(Node):

    def __init__(self):

        #constructor del nodo
        super().__init__("subscriber")

        #creación del suscriptor
        self.subscriber_ = self.create_subscription(String, "informacion", self.subscriber_cb, 10)

    #callback que se ejecuta cada vez que llega un msj al tópico
    def subscriber_cb (self, msg: String):
        self.get_logger().info(msg.data)

#función principal
def main(args=None):

    #inicializa ros 2, crea el nodo y lo mantiene activo
    rclpy.init(args=args)
    node = Subscriber()
    rclpy.spin(node)
    rclpy.shutdown

if __name__ == "__main__":

    main()

File: modulo1/src/ros2_ws/src/fulgor_cpp_examples/src/subscriber.cpp
//Descripción: Nodo suscriptor del tópico /información que recibe mensajes del tipo String.

#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::placeholders; 

//define una clase Subscriber (crea un nodo) que hereda de rclcpp::Node
class Subscriber: public rclcpp::Node 
{
public:
    //constructor
    Subscriber(): Node ("subscriber")
    {
        //creación del subscriber
        subscriber_ = this ->create_subscription<example_interfaces::msg::String>("informacion", 10, 
        std::bind(&Subscriber::cbSubscriberInformacion, this,_1));

        RCLCPP_INFO(this->get_logger(), "Nodo creado correctamente!");

    }
private: 

    //callback que se ejecuta cuando llega un nuevo mensaje al tópico
    void cbSubscriberInformacion(const example_interfaces::msg::String::SharedPtr msg)
    {
        //imprime el mensaje del tópico en pantalla
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
    }

    //suscripción guardado como SharedPtr para que no se destruya al salir del constructor
    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

//función principal
int main (int argc, char **argv)
{
    //inicializa ros 2, crea el nodo y lo mantiene activo
    rclcpp::init(argc, argv);
    auto node = std::make_shared<Subscriber>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
